.github/workflows/lint.yml

```diff
@@ -0,0 +1,27 @@
+name: Lint
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  golangci-lint:
+    name: Run on Ubuntu
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone the code
+        uses: actions/checkout@v4
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '~1.22'
+
+      - name: Install golangci-lint
+        run: |
+          curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh \
+            | sh -s -- -b $(go env GOPATH)/bin v1.63.4
```

Comment:
Hard coding the version "v1.63.4" is not good, as we will need to manually maintain it. 

Please follow https://github.com/etcd-io/etcd/blob/ad11af73aee38babfd7963aa556fa8d5372e7055/tools/mod/go.mod#L13 
cc @ivanvc @jmhbnz